lib/virt_autotest/utils.pm

```diff
@@ -42,62 +42,89 @@ sub trim {
-        my $_libvirtd_pid = script_output(q@ps -ef |grep [l]ibvirtd | gawk '{print $2;}'@);
-        my $_libvirtd_cmd = script_output("ps -o command $_libvirtd_pid | tail -1");
-        assert_script_run("kill -9 $_libvirtd_pid");
-        assert_script_run("$_libvirtd_cmd");
-    }
-    else {
-        systemctl("restart libvirtd", timeout => 180);
-    }
-    record_info("Libvirtd has been restarted!");
-    save_screenshot;
-}
-
 #return 1 if test is expected to run on XEN hypervisor
 sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
+# Usage: check_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
+# to specify daemons which will be checked: check_modular_libvirt_daemons(qemu storage ...)
+# to check all required modular daemons without any daemons passed
+sub check_modular_libvirt_daemons {
+    my @daemons = @_;
+
+    if (!@daemons) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy log lock);
+        (is_xen_host) ? push @daemons, 'xen' : push @daemons, 'qemu';
+    }
+
+    foreach my $daemon (@daemons) {
+        systemctl("status virt${daemon}d.service");
+        if (($daemon == 'lock') || ($daemon == 'log')) {
+            systemctl("status virt${daemon}d\{,-admin\}.socket");
+        } else {
+            systemctl("status virt${daemon}d\{,-ro,-admin\}.socket");
+        }
+    }
+
+    record_info("Modular Libvirt daemons checked", join(' ', @daemons));
+    save_screenshot;
+}
+
 # Usage: restart_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
 # to specify daemons which will be restarted: restart_modular_libvirt_daemons(virtqemud virtstoraged ...)
 # to restart all modular daemons without any daemons passed
 sub restart_modular_libvirt_daemons {
-    my @daemons;
+    my @daemons = @_;
 
-    if (@_ == 0) {
-        if (is_xen_host) {
-            @daemons = qw(virtxend virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd);
-        } else {
-            @daemons = qw(virtqemud virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd);
-        }
-    } else {
-        @daemons = @_;
+    if (!@daemons) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy log lock);
+        (is_xen_host) ? push @daemons, 'xen' : push @daemons, 'qemu';
     }
 
     if (is_alp) {
         record_soft_failure("Restarting modular libvirt daemons has not been implemented in ALP. See poo#129086");
     } else {
         # Restart the sockets first
         foreach my $daemon (@daemons) {
-            systemctl("restart $daemon\{,-ro,-admin\}.socket");
+            if (($daemon == 'lock') || ($daemon == 'log')) {
+                systemctl("restart virt${daemon}d\{,-admin\}.socket");
+            } else {
+                systemctl("restart virt${daemon}d\{,-ro,-admin\}.socket");
+            }
```

Comment:
I found there is ```virtproxyd-ro.socket``` on leap15.4, so I think it should also exists on 15-SP6.

Reply:
Awesome! Good point! we should create new ticket to trace it. And then,  make sure this situation on our LEAP with modular libvirt as new behavior or new product bug. 

back to this PR, which just only adapt SLE automation to modular libvirt as the purpose without openSUSE LEAP. mean that, focus on SLE15SP6 with modular libvirt.